src/lib/tpm.c

```diff
@@ -2478,6 +2478,53 @@ static CK_RV tpm_object_data_populate_ecc(TPM2B_PUBLIC *out_pub, tpm_object_data
     return rv;
 }
 
+static CK_RV check_tpm_ecc(ESYS_CONTEXT *context, tpm_key_data tpmdat) {
+
+    /* get ecc-curve capability from TPM*/
+    TPMU_CAPABILITIES *capabilities = NULL;
```

Comment:
move down to usage

Reply:
If I move it down, I would put it into the "tpm2_generate_key" function, where the "check_tpm_ecc" is used. That would not work, would it?